docker-compose.yml

```diff
@@ -0,0 +1,210 @@
+      - shared_assets_dir:/app/oppia/assets
+      - shared_webpack_bundles:/app/oppia/webpack_bundles
+    healthcheck:
+      test: test "$(ls -A /app/oppia/webpack_bundles)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    entrypoint: node
+    command:
+      - "/app/oppia/node_modules/webpack/bin/webpack.js"
+      - "--config"
+      - "/app/oppia/webpack.dev.config.ts"
+      - "--watch"
+      - "--color"
+      - "--progress"
+
+  google_app_engine:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-app-engine
+    ports:
+      - "8181:8181"
+      - "8000:8000"
+    volumes:
+      - .:/app/oppia
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_third_party:/app/oppia/third_party
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+      - shared_assets_dir:/app/oppia/assets
+      - shared_angular_build:/app/oppia/dist
+      - shared_webpack_bundles:/app/oppia/webpack_bundles
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+      - oppia_is_dockerized=true
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8181/ || exit 1
+      timeout: 10s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/dev_appserver.py
+    command: 
+      - "/app/oppia/app_dev_docker.yaml"
+      - "--runtime"
+      - "python38"
+      - "--host=0.0.0.0"
+      - "--port=8181"
+      - "--admin_host=0.0.0.0"
+      - "--admin_port=8000"
+      - "--skip_sdk_update_check=True"
+
+  datastore:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/gcloud
+    command:
+      - "beta"
+      - "emulators"
+      - "datastore"
+      - "start"
+      - "--project"
+      - "dev-project-id"
+      - "--data-dir"
+      - "/app/cloud_datastore_emulator_cache"
+      - "--host-port=0.0.0.0:8089"
+      - "--consistency=1.0"
+      - "--quiet"
+      - "--no-store-on-disk"
+
+  firebase:
+    image: node@sha256:48650ae5c5dc9fbd0d0559fd1248e779f7214ba99434420a089bc7853aeefb65
+    container_name: oppia-firebase-emulator
+    ports:
+      - "9099:9099"
+      - "4000:4000"
+    volumes:
+      - ./.firebase.json:/app/oppia/.firebase.json
+      - shared_node_modules:/app/oppia/node_modules
```

Comment:
How node_modules are downloaded? I don't see this container depending on any other services, so how is it downloading node_modules?

Possible solution:
- Add depends_on webpack
OR
- Create separate `FROM node as firebase` and install the required firebase dep using `npm install firebase.x.y.z` and add an entrypoint.

Reply:
node_modules are within shared vol. note that the shared volumes are created at the time when the containers are in "creating" stage and are not started. so no need to add depends_on. we have all required deps for firebase downloaded within the node_modules

Reply:
If I'll run `docker compose up firebase`, I'm assuming it will fail because node_modules doesn't exist, right? You can make it work using any possible solution I've provided above or if you have any other ideas?

Reply:
yes @DubeySandeep good catch, the service will fail as it is depending on the node_modules shared vol which is created within the angular-build/webpack service. should I add a depends_on condition.
also, note that we need to have depends_on condition for firebase service as we need to run this service in the stage 2 (i.e. after webapck and ng-build service are up)

Reply:
I'm fine with that but would have preferred having an independent container as it can be run independently, just add a TODO comment to make this service independent as it only needs firebase-related dep. to start.

Reply:
done